py/desitarget/secondary.py

```diff
@@ -508,7 +508,8 @@ def match_secondary(primtargs, scxdir, scndout, sep=1.,
     if np.any(scnd_update):
         # APC Allow changes to primaries if the DESI_TARGET bitmask has
         # APC only the following bits set, in any combination.
-        update_from_scnd_bits = desi_mask['MWS_ANY'] & desi_mask['STD_BRIGHT'] & desi_mask['STD_FAINT'] & desi_mask['STD_WD']
+        log.info('Testing if secondary targets can update {} matched primaries'.format(scnd_update.sum()))
+        update_from_scnd_bits = desi_mask['SCND_ANY'] | desi_mask['MWS_ANY'] | desi_mask['STD_BRIGHT'] | desi_mask['STD_FAINT'] | desi_mask['STD_WD']
```

Comment:
Thanks for catching this general `&` instead of `|` bug.

My only concern: Doesn't this mean that _all_ secondary targets are allowed to update the priorities of the primaries? 

I thought we'd discussed that it was OK to update the priorities for primaries if they were _just_ MWS targets? This seems like any primary object that has `SCND_ANY` set will be allowed to trump primaries, which is dangerous for large-scale-structure targets that match a secondary. What would happen for a target with, say, just `LRG` and `SCND_ANY` set?

Reply:
My intention was to allow updates to a target with SCND_ANY set ONLY if that target **does not** have any other bits (e.g. LRG) set, *except* the specified MW bits. As written, I think this is what happens. .

The important thing is how that mask gets used in the next line, `scnd_update &= ((targets[desi_target] & ~update_from_scnd_bits) == 0)`. The (A &~B) is true if any bit is set in A that is *not* set in B, so checking `== 0` leaves the complement (no bit set in A that isn't set in B). 

Maybe this could be written in a less abstruse way, but I think it works as it is. Without the SCND_ANY in the 'updatable' mask, this logic would not update any secondary targets

For example:

```python
test_target = desi_mask['LRG'] | desi_mask['SCND_ANY']
print('Target bits:', test_target)
scnd_update = (test_target & desi_mask['SCND_ANY']) != 0
print('Is secondary:', scnd_update)
update_from_scnd_bits = (desi_mask['SCND_ANY'] | desi_mask['MWS_ANY'] | desi_mask['STD_BRIGHT'] | desi_mask['STD_FAINT'] | desi_mask['STD_WD'])
scnd_update &= ((test_target & ~update_from_scnd_bits) == 0)
print('Is updatable:', scnd_update)
```
```
Target bits: 4611686018427387905
Is secondary: True
Is updatable: False
```
And the positive case:
```python
test_target = desi_mask['MWS_ANY'] | desi_mask['STD_BRIGHT'] | desi_mask['SCND_ANY']
print('Target bits:', test_target)
scnd_update = (test_target & desi_mask['SCND_ANY']) != 0
print('Is secondary:', scnd_update)
update_from_scnd_bits = (desi_mask['SCND_ANY'] | desi_mask['MWS_ANY'] | desi_mask['STD_BRIGHT'] | desi_mask['STD_FAINT'] | desi_mask['STD_WD'])
scnd_update &= ((test_target & ~update_from_scnd_bits) == 0)
print('Is updatable:', scnd_update)
```
```
Target bits: 6917529062000820224
Is secondary: True
Is updatable: True
```

Reply:
I confess, you did confuse me with the `&` of a -ve bit number. But, I've tested this again and it seems to work. Thanks for the informative demonstration, here.